airbyte-ci/connectors/pipelines/pipelines/models/secrets.py

```diff
@@ -0,0 +1,138 @@
+            # 1 means enabled version
+            if version.state == 1:
+                request = secretmanager_v1.AccessSecretVersionRequest(
+                    name=version.name,
+                )
+                response = self.gsm_client.access_secret_version(request=request)
+                return response.payload.data.decode()
+        raise SecretNotFoundError(f"No enabled secret version in GSM found for secret {name}")
+
+
+class EnvVarSecretStore(SecretStore):
+    def _fetch_secret(self, name: str) -> str:
+        try:
+            return os.environ[name]
+        except KeyError:
+            raise SecretNotFoundError(f"The environment variable {name} is not set.")
+
+
+class LocalDirectorySecretStore(SecretStore):
+    def __init__(self, local_directory_path: Path) -> None:
+        if not local_directory_path.exists() or not local_directory_path.is_dir():
+            raise ValueError(f"The path {local_directory_path} does not exists on your filesystem or is not a directory.")
+        self.local_directory_path = local_directory_path
+
+    def _fetch_secret(self, name: str) -> str:
+        secret_path = self.local_directory_path / name
+        if not secret_path.exists():
+            raise SecretNotFoundError(f"The file {secret_path} does not exists.")
+        if not secret_path.is_file():
+            raise SecretNotFoundError(f"The path {secret_path} is not a file.")
+        return secret_path.read_text()
+
+    def get_all_secrets(self) -> List[Secret]:
+        return [
+            Secret(
+                name=str(file.relative_to(self.local_directory_path)),
+                secret_store=self,
+                file_name=str(file.relative_to(self.local_directory_path)),
+            )
+            for file in self.local_directory_path.rglob("*")
+            if file.is_file()
+        ]
+
+
+class InMemorySecretStore(SecretStore):
+    def __init__(self) -> None:
+        self._store: Dict[str, str] = {}
+
+    def add_secret(self, name: str, value: str) -> Secret:
+        self._store[name] = value
+        return Secret(name, self)
+
+    def _fetch_secret(self, name: str) -> str:
+        try:
+            return self._store[name]
+        except KeyError:
+            raise SecretNotFoundError(f"Secret named {name} can't be found in the in memory secret store")
+
+
+@dataclass
+class Secret:
+    name: str
+    secret_store: SecretStore
+    file_name: str | None = None
+
+    def __post_init__(self) -> None:
+        self.value: str = self.secret_store.fetch_secret(self.name)
+        self.value_hash: str = self._get_value_hash(self.value)
+
+    @staticmethod
+    def _get_value_hash(value: str) -> str:
+        byte_string = value.encode("utf-8")
+        md5_hash = hashlib.md5()
+        md5_hash.update(byte_string)
+        return md5_hash.hexdigest()[:20]
+
+    def as_dagger_secret(self, dagger_client: DaggerClient) -> DaggerSecret:
+        return dagger_client.set_secret(self.value_hash, self.value)
```

Comment:
❓ Why do we have to transform it to a md5 hash?

Reply:
@bnchrch : 
* The Dagger client is shared between connector pipelines
* Dagger keeps an internal in memory store of secrets, by name
* If two connector use the same secret name they would step on each others toe and we'd have strange errors in CI to debug.
* I could have used a namespaced secret name like `connector-name_secret_name` but I believe these new model should be available for non connector use cases.
* MD5 hashing of sounded like the simplest and straightforward way of getting unique secret names